fb_apt/providers/sources_list.rb

```diff
@@ -0,0 +1,70 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+action :run do
+  repos = []
+  mirror = node['fb_apt']['mirror']
+  distro = node['lsb']['codename']
+
+  # only add base repos if mirror is set and codename is available
+  if mirror && distro
+    components = 'main'
```

Comment:
it's an list, so instead of managing spaces manually which will break at some point with some diff, make it an array, and when you use it, call join